source/mainnet/smart-contracts/tutorials/nft-marketplace/smart-contract.rst

```diff
@@ -0,0 +1,253 @@
+
+.. code-block:: console
+
+    cargo concordium build --out dist/marketplace-contract/module.wasm.v1 --schema-out dist/marketplace-contract/schema.bin
+
+.. image:: images/build-sc.png
+    :width: 75%
+
+Deploy the compiled Wasm file with the following command.
+
+.. code-block:: console
+
+    concordium-client module deploy dist/marketplace-contract/module.wasm.v1 --sender <YOUR-ACCOUNT-ADDRESS> --name <YOUR-CONTRACT-NAME> --grpc-port 10001
+
+.. image:: images/deploy-sc.png
+    :width: 75%
+
+Initialize the marketplace smart contract
+=========================================
+
+Create an instance of the marketplace contract with the command below. Look at the ``init`` function. As you can see in the code below, it expects an ``InitParam`` input which will be used to set the commission value in the ``State::new()`` function while initializing your marketplace state.
+
+.. code-block:: Rust
+
+    /// Initializes a new Marketplace Contract
+    ///
+    /// This function can be called by using InitParams.
+    /// The commission should be less than the maximum allowed value of 10000 basis points
+    #[init(contract = "Market-NFT", parameter = "InitParams")]
+    fn init<S: HasStateApi>(
+        ctx: &impl HasInitContext,
+        state_builder: &mut StateBuilder<S>,
+    ) -> InitResult<State<S, ContractTokenId, ContractTokenAmount>> {
+        let params: InitParams = ctx
+            .parameter_cursor()
+            .get()
+            .map_err(|_e| MarketplaceError::ParseParams)?;
+
+        if params.commission > MAX_BASIS_POINTS {
+            return InitResult::Err(Reject::from(MarketplaceError::InvalidCommission));
+        }
+
+        Ok(State::new(state_builder, params.commission))
+    }
+
+Run the following command to initialize the contract.
+
+.. code-block:: console
+
+    concordium-client contract init market --contract Market-NFT --parameter-json ../sample-artifacts/marketplace/init.json --sender <YOUR-ADDRESS> --energy 3000 --schema dist/marketplace-contract/schema.bin --grpc-port 10001
+
+.. image:: images/init-sc.png
+    :width: 75%
+
+UpdateOperator and add token
+============================
+
+Now you are going to sell your NFT. In order to do that, you need to tell your token “I’m going to approve transferring you to a particular address.” You do this with ``updateOperator()`` function in CIS-2, which is standard function. What you need to be careful about is that you need to call your token contract with its schema. Then you should be able to do an ``add()`` operation. Go to the sample artifacts and ``update-operator.json`` to change the marketplace address. You are giving your token’s temporary custody to the marketplace contract. If you want to test the logic behind it, skip this and try to call ``add()`` directly from the contract. It won't work because your token’s state is not updated yet. As you can see below, the ``UpdateOperator()`` function updates the state of the token.
```

Comment:
I don't think the description of "I'm going to approve you ..." is accurate.

I'm guessing you are trying to convey that the marketplace will be transferring the token. So you need to give it permission to do so.

Reply:
```You are giving your token’s temporary custody to the marketplace contract. ```

That does not read right. You are allowing the marketplace contract to act on your behalf. But you can still do it and you still own the token. I think "giving custody" reads wrong. But perhaps @dg-concordium will correct me.